pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -771,6 +804,11 @@ func (c *Controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.
 
 		c.updateEDS(ep, event)
 
+		instances := c.getServiceInstanceByEndpoint(*ep)
+		for _, instance := range instances {
+			f(instance, event)
```

Comment:
@zcahana @howardjohn @rshriram This may duplicate with L809 and also not that efficient, would rather suggest having a careful look at #16338.